libs/agno/agno/agent/agent.py

```diff
@@ -4243,66 +4383,78 @@ def get_tool_call_history(num_calls: int = 3) -> str:
 
         return get_tool_call_history
 
-    def search_knowledge_base(self, query: str) -> str:
-        """Use this function to search the knowledge base for information about a query.
+    def search_knowledge_base_function(self, async_mode: bool = False) -> Callable:
+        """Factory function to create a search_knowledge_base function with filters."""
+        # Determine which filters to use
+        effective_filters = self.effective_knowledge_filters
```

Comment:
The effective filters should be calculated on each run. Definitely not assigned globally.